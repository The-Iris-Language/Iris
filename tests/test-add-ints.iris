.. Tests adding ints and assigning to variables or class vars

class Dog () {
    public:
        int age;
}

class Main () {
    public: 
        int univ main() {
            int a = 1 + 2;
            Olympus.print("a = 3");
            a = -1 + 2;
            Olympus.print("a = 1");
            a = -5 + a;
            Olympus.print("a = -4");
            a = -5 + -6;
            Olympus.print("a = -11");
            Dog d;
            d.age = 1 + 1;
            Olympus.print("d.age = 2");
            d.age = -5 + -5;
            Olympus.print("d.age = -10");
            d.age = a + a; 
            Olympus.print("d.age = -22");
            a = d.age + 1;
            Olympus.print("a = -21");
            return 0;
        }
}