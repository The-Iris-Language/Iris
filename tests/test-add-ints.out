a = 3
a = 1
a = -4
a = -11
d.age = 2
d.age = -10
d.age = -22
a = -21
